tests/unit/test_application.py

```diff
@@ -1121,29 +1350,88 @@ def grammar_project(tmp_path):
     project_file.write_text(contents)
 
 
+@pytest.fixture()
+def non_grammar_project_full(tmp_path):
+    """A project that builds on amd64 to riscv64."""
+    project_file = tmp_path / "testcraft.yaml"
+    project_file.write_text(FULL_PROJECT_YAML)
+
+
+@pytest.fixture()
+def grammar_project_full(tmp_path):
+    """A project that builds on amd64 to riscv64 and s390x."""
+    project_file = tmp_path / "testcraft.yaml"
+    project_file.write_text(FULL_GRAMMAR_PROJECT_YAML)
+
+
+@pytest.fixture()
+def non_grammar_build_plan(mocker):
+    """A build plan to build on amd64 to riscv64."""
+    host_arch = "amd64"
+    base = util.get_host_base()
+    build_plan = [
+        models.BuildInfo(
+            f"platform-{build_for}",
+            host_arch,
+            build_for,
+            base,
+        )
+        for build_for in ("riscv64",)
```

Comment:
I know Alex just suggested this change but here's another suggestion:
```suggestion
            "platform-riscv64",
            host_arch,
            build_for,
            base,
        )
```